src/twisted/python/threadpool.py

```diff
@@ -9,15 +9,33 @@
 instead of creating a thread pool directly.
 """
 
+from __future__ import annotations
+
 from threading import Thread, current_thread
-from typing import List
+from typing import Any, Callable, List, Optional, TypeVar
+
+from typing_extensions import ParamSpec, Protocol, TypedDict
 
 from twisted._threads import pool as _pool
 from twisted.python import context, log
 from twisted.python.deprecate import deprecated
 from twisted.python.failure import Failure
 from twisted.python.versions import Version
 
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+
+class _SupportsQsize(Protocol):
+    def qsize(self) -> int:
+        pass
+
+
+class _State(TypedDict):
```

Comment:
Is it expected to define types without docstrings?

Is this the convention for python typing?

Reply:
Types should really have docstrings just like anything else, but sometimes you need to describe a very simple object like "has a `qsize` method" and bloating it up with multiple lines of docstrings does not facilitate any understanding.

I'd prefer if we used `...` in these situations instead of `pass` though, since that's the convention for stubs.

Reply:
thanks for the info.